hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java

```diff
@@ -198,6 +198,8 @@ public TransactionInfo runAllChecks(
         // 1. Check the syntax
         final var maxBytes = maxIngestParseSize(configuration);
         final var txInfo = transactionChecker.parseAndCheck(serializedTransaction, maxBytes);
+        // check jumbo size after parsing
+        transactionChecker.checkJumboTransactionBody(txInfo);
```

Comment:
The downside of checking for jumbo size inside of `runAllChecks` is that we start to handle specific transaction functionality inside of the very-generic ingestion algorithm. 

Even better than passing in `maxBytes` would be to pass in the config itself, and let `transactionChecker` determine what it needs from the config:

```suggestion
        final var txInfo = transactionChecker.parseAndCheck(serializedTransaction, configuration);
```

Then the implementation of checking the txn size is encapsulated inside of `transactionChecker` with the other parsing-related conditions. Looking at the current code (without these changes), I'm already asking myself why the size of the transaction is handled separately, especially underneath a comment that says `// 1. Check the syntax`. 

Thoughts?

Reply:
In `transactionParser`, we perform a generic transaction size check: if the `jumboFlag` is enabled, the limit is 130KB; otherwise, it’s 6KB. At this stage, we don’t yet know the `HederaFunctionality`, so this check is functionality-agnostic. In the jumbo HIP, we introduced an additional size check after parsing the transaction, which enforces size limits based on the specific functionality.

Reply:
Ah I see, you're right, it's not specific; just the length of the message 👍